drivers/retained_mem/Kconfig.nrf

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config RETAINED_MEM_NRF_GPREGRET
+	bool "nRF GPREGRET driver"
+	default y
```

Comment:
Why do you want to have it y by default?

Reply:
This is how all drivers work, if you build for an nrf52840dk and enable i2c, it will automatically select the nordic i2c driver